fdbserver/workloads/ConsistencyCheck.actor.cpp

```diff
@@ -1142,7 +1159,14 @@ struct ConsistencyCheckWorkload : TestWorkload {
 		}
 
 		if (foundExtraDataStore) {
-			wait(delay(10)); // let the cluster get to fully_recovered after the reboot before retrying
+			// Let the cluster fully recover after rebooting/killing storage servers with extra stores.
+			//
+			// This requires an end-to-end comitting transaction to ensure recovery has started in case
+			// any stateless processes, like the commit proxy, were killed.
+			wait(::success(doEmptyCommit(cx)));
```

Comment:
why is waiting the empty commit needed, since you are waiting on `FULLY_RECOVERED` state below anyway?

Reply:
If you just check the recovery state without running a transaction, you may be working with the recovery state from a previous generation. Waiting for a successful transaction makes sure you are up to date.